tsinfer/formats.py

```diff
@@ -1356,7 +1356,9 @@ def subset(self, individuals=None, sites=None, **kwargs):
         sites = set(sites)
         if len(sites) != num_sites:
             raise ValueError("Duplicate site IDS")
-        with SampleData(sequence_length=self.sequence_length, **kwargs) as subset:
+        if "sequence_length" not in kwargs:
```

Comment:
What about adding ``sequence_length=None`` to the args above, and then:
```
sequence_length = sequence_length if sequence_length is not None else self.sequence_length
```
Manually looking into kwargs should be a last-resort option. (Should make sequence_length kw only though)

Reply:
Done